pkg/function/backups_info.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	sp "k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes"
+
+	kanister "github.com/kanisterio/kanister/pkg"
+	"github.com/kanisterio/kanister/pkg/format"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/param"
+	"github.com/kanisterio/kanister/pkg/restic"
+)
+
+const (
+	// BackupsInfoNamespaceArg provides the namespace
+	BackupsInfoNamespaceArg = "namespace"
+	// BackupsInfoArtifactPrefixArg provides the path to restore backed up data
+	BackupsInfoArtifactPrefixArg = "backupArtifactPrefix"
+	// BackupsInfoEncryptionKeyArg provides the encryption key to be used for deletes
+	BackupsInfoEncryptionKeyArg = "encryptionKey"
+	// BackupsInfoPodOverrideArg contains pod specs to override default pod specs
+	BackupsInfoPodOverrideArg    = "podOverride"
+	BackupsInfoJobPrefix         = "get-backups-info-"
+	BackupsInfoFileCount         = "fileCount"
+	BackupsInfoSize              = "size"
+	BackupsInfoSnapshotIDs       = "snapshotIDs"
+	BackupsInfoPasswordIncorrect = "passwordIncorrect"
+	BackupsInfoRepoUnavailable   = "repoUnavailable"
+)
+
+func init() {
+	kanister.Register(&BackupsInfoFunc{})
+}
+
+var _ kanister.Func = (*BackupsInfoFunc)(nil)
+
+type BackupsInfoFunc struct{}
```

Comment:
Not a fan of the function name. @DeepikaDixit @tdmanv - would `DescribeBackups` be better?

Reply:
that works for me

Reply:
👍